third_party/ruby/sorbet_ruby_2_7.patch

```diff
@@ -2667,3 +2667,333 @@ index 79a6cf014ec85..8596e416d7b7a 100644
+ }
+ 
+@@ -265,12 +264,11 @@ setup_method_cfunc_struct(rb_method_cfunc_t *cfunc, VALUE (*func)(), int argc)
+ }
+ 
+ static void
+-setup_method_sorbet_struct(rb_method_sorbet_t *sorbet, rb_sorbet_func_t func, const rb_sorbet_param_t *param, rb_iseq_t *iseqptr, rb_cref_t *cref)
++setup_method_sorbet_struct(rb_method_sorbet_t *sorbet, rb_sorbet_func_t func, const rb_sorbet_param_t *param, rb_iseq_t *iseqptr)
+ {
+     sorbet->func = func;
+     sorbet->param = param;
+     sorbet->iseqptr = iseqptr;
+-    sorbet->cref = cref;
+ }
+ 
+ MJIT_FUNC_EXPORTED void
+@@ -309,7 +307,7 @@ rb_method_definition_set(const rb_method_entry_t *me, rb_method_definition_t *de
+ 	  case VM_METHOD_TYPE_SORBET:
+ 	    {
+ 		rb_method_sorbet_t *sorbet = (rb_method_sorbet_t *)opts;
+-		setup_method_sorbet_struct(UNALIGNED_MEMBER_PTR(def, body.sorbet), sorbet->func, sorbet->param, sorbet->iseqptr, sorbet->cref);
++		setup_method_sorbet_struct(UNALIGNED_MEMBER_PTR(def, body.sorbet), sorbet->func, sorbet->param, sorbet->iseqptr);
+ 		return;
+ 	    }
+ 	  case VM_METHOD_TYPE_ATTRSET:
+
+From 46b9109c7ddcaa4e8bbc6b01af83357d966f4f43 Mon Sep 17 00:00:00 2001
+From: Nathan Froyd <froydnj@gmail.com>
+Date: Wed, 29 Sep 2021 17:13:43 -0400
+Subject: [PATCH 28/28] pass the call data to sorbet method functions (#3)
+
+---
+ method.h        | 2 +-
+ vm_eval.c       | 2 +-
+ vm_insnhelper.c | 2 +-
+ 3 files changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/method.h b/method.h
+index 75cc5785d25a..294ed792843b 100644
+--- a/method.h
++++ b/method.h
+@@ -199,7 +199,7 @@ typedef struct rb_sorbet_param_struct {
+     const ID *kw_table;
+ } rb_sorbet_param_t;
+ 
+-typedef VALUE (*rb_sorbet_func_t)(int, VALUE *, VALUE, struct rb_control_frame_struct *);
++typedef VALUE (*rb_sorbet_func_t)(int, VALUE *, VALUE, struct rb_control_frame_struct *, void *);
+ 
+ typedef struct rb_method_sorbet_struct {
+     /* cf. rb_method_cfunc_struct, but we only support one argument style */
+diff --git a/vm_eval.c b/vm_eval.c
+index 52805d317f02..26d3d8906f3f 100644
+--- a/vm_eval.c
++++ b/vm_eval.c
+@@ -144,7 +144,7 @@ vm_call0_sorbet_with_frame(rb_execution_context_t* ec, struct rb_calling_info *c
+                                                     0, reg_cfp->sp, sorbet->iseqptr->body->local_table_size, sorbet->iseqptr->body->stack_max);
+ 
+         /* TODO: eventually we want to pass cd in here to assist with kwargs parsing */
+-        val = (*sorbet->func)(argc, argv, recv, new_cfp);
++        val = (*sorbet->func)(argc, argv, recv, new_cfp, cd);
+ 
+ 	CHECK_CFP_CONSISTENCY("vm_call0_sorbet_with_frame");
+ 	rb_vm_pop_frame(ec);
+diff --git a/vm_insnhelper.c b/vm_insnhelper.c
+index a891794d0a08..05171ad04ca7 100644
+--- a/vm_insnhelper.c
++++ b/vm_insnhelper.c
+@@ -2622,7 +2622,7 @@ vm_call_sorbet_with_frame_normal(rb_execution_context_t *ec, rb_control_frame_t
+ 
+     reg_cfp->sp -= argc + 1;
+     /* TODO: eventually we want to pass cd in here to assist with kwargs parsing */
```

Comment:
Same, should we remove these TODOs?

Reply:
Yeah, we should eventually remove these.  I will take a note to do that.